lib/streamlit/navigation/page.py

```diff
@@ -114,17 +122,26 @@ def __init__(
         inferred_icon = ""
         if isinstance(page, Path):
             inferred_icon, inferred_name = page_icon_and_name(page)
+        elif hasattr(page, "__name__"):
+            inferred_name = str(page.__name__)
         elif title is None:
-            if hasattr(page, "__name__"):
-                inferred_name = str(page.__name__)
-            else:
-                raise StreamlitAPIException(
-                    "Cannot infer page title for Callable. Set the `title=` keyword argument."
-                )
+            raise StreamlitAPIException(
+                "Cannot infer page title for Callable. Set the `title=` keyword argument."
+            )
 
         self._page: Path | Callable[[], None] = page
         self._title: str = title or inferred_name.replace("_", " ")
         self._icon: str = icon or inferred_icon
+        if url_path is not None and url_path.strip() == "" and not default:
+            raise StreamlitAPIException(
+                "The URL path cannot be an empty string unless the page is the default page."
+            )
+
+        self._url_path: str = (
+            (url_path and url_path.lstrip("/"))
```

Comment:
nit (probably personal opinion): `url_path and url_path.lstrip("/")` might be a common React-syntax (with `&&`), but I have not seen it that often in Python. It might be more clear to write
`url_path.lstrip("/") if url_path else inferred_name if inferred_name else self.title.replace(" ", "_")`
Though I am not necessarily saying that it is easier to read or so, just wanted to express that this syntax here might not be too common (unless it is and I am not aware of it 🙂)

Reply:
I provided a different approach. I realized that the title should never be available because the `inferred_name` will always have a string value (from the file or from `__name__`), so defaulting to the `inferred_name` and overriding it if the url_path is specified seems like a cleaner approach.